### ISSUE TITLE
Occasional SIGBUS error 

### ISSUE THREAD

**Issue by chcokr** [shlee0605/proxy#6]
Every once is a few minutes, I get a SIGBUS error on the following line while browsing:

https://github.com/shlee0605/proxy/blob/763a52af20e8874e514ee0fb5ae7e1f2b1f20769/http_get.c#L60

Can you take a look at what's causing this issue?

**Comment by shlee0605** [comment]
```
while (*crlf_loc != '\r' || *(crlf_loc + 1) != '\n') {
    crlf_loc += 1;
}
```
I got the `SIGSEGV` error from the line you mentioned above.
I think the value of `crlf_loc` goes beyond the `header_str`'s allocated memory index. 
I tried the following code and it removes `SIGSEGV`error. However, parsing parts now does not work properly.
```
while (*crlf_loc != '\r' || *(crlf_loc + 1) != '\n') {
    crlf_loc += 1;
}
if(crlf_loc > header_str + sizeof(header_str)) {
      break;
}
```

**Comment by shlee0605** [comment]
fixed #4
